pkg/protocol/engine/tipmanager/v1/tipmanager.go

```diff
@@ -150,6 +199,40 @@ func (t *TipManager) setupBlockMetadata(tipMetadata *TipMetadata) {
 	t.blockAdded.Trigger(tipMetadata)
 }
 
+// trackLatestValidationBlock tracks the latest validator block and takes care of marking the corresponding TipMetadata.
+func (t *TipManager) trackLatestValidationBlock(tipMetadata *TipMetadata) (teardown func()) {
+	if _, isValidationBlock := tipMetadata.Block().ValidationBlock(); !isValidationBlock {
+		return
+	}
+
+	committee, exists := t.retrieveCommitteeInSlot(tipMetadata.Block().ID().Slot())
```

Comment:
This is a hot path that reads data from the committee KVStore for each block - we should add a cache at some level. Maybe we should have a committee cache in SeatManager and clear it whenever SybilProtection.Reset() is called? Not necessarily as part of this PR, but then we should create an issue for that.